content/en/docs/advanced-tasks/adding-admission-and-conversion-webhooks.md

```diff
@@ -0,0 +1,138 @@
+linkTitle: "Admission and Conversion Webhooks"
+weight: 3
+---
+
+## The WebhookDefinition ClusterServiceVersion Object
+
+Operators may define [validating](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook), [mutating](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook), and [conversion](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#webhook-conversion) webhooks within its [ClusterServiceVersion resource (CSV)](/docs/Concepts/crds/clusterserviceversion).
+
+These webhooks are defined with the CSV's [WebhookDefinition](https://github.com/operator-framework/api/blob/7856a40f92893fe94d19d223f5277d1d116ffc67/pkg/operators/v1alpha1/clusterserviceversion_types.go#L164-L180) array. The `WebhookDescription` object contains a union of the fields defined in the [ValidatingWebhookConfiguration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#validatingwebhookconfiguration-v1-admissionregistration-k8s-io) and [MutatingWebhookConfiguration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#mutatingwebhookconfiguration-v1-admissionregistration-k8s-io) type with the exception of the NamespaceSelector, which is generated by OLM to match namespaces scoped by the [OperatorGroup](./operator-scoping-with-operatorgroups.md) that the operator is deployed in.
+
+In addition to these fields, OLM requires that you define:
+
+- The `Type` of the Webhook, which must be set to `ValidatingAdmissionWebhook`, `MutatingAdmissionWebhook`, or `ConversionWebhook`
+- The `DeploymentName` that OLM must mount the CA Cert information into, this should match the name of a deployment defined in the CSV
+
+## Deploying an operator with webhooks using OLM
+
+### OLM Requirements for Operators with Webhooks
+
+OLM has a few constraints that must be considered when developing an operator featuring validating, mutating, or conversion webhooks.
+
+#### Certificate Authority Requirements
+
+OLM is configured to create a single Certificate Authority (CA) for each webhook deployment. The logic that generates and mounts the CA into the deployment was originally used by the [API Service](https://kubernetes.io/docs/tasks/access-kubernetes-api/setup-extension-api-server/) lifecycle logic. As a result:
+
+- The tls Cert file will be mounted to the deployment at `/apiserver.local.config/certificates/apiserver.crt`
+- The tls Key file will be mounted to the deployment at `/apiserver.local.config/certificates/apiserver.key`
+
+If the operator was built with [Kubebuilder](https://book.kubebuilder.io/) or the [Operator-SDK](https://github.com/operator-framework/operator-sdk), [this commit](https://github.com/awgreene/webhook-operator/commit/3ef33a1b7090180bd873fb2ecbcddb9a92a79f0b) can be used as a reference to update where the operator's webhook expects the Certs mounted by OLM.
```

Comment:
Should we explain everything in plain text instead of referencing a code PR here?

Reply:
We have a BZ to make it support sdk defaults automatically so I'll remove this link all together eventually.